src/Microsoft.AspNetCore.Cors/Infrastructure/UriHelpers.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Cors.Infrastructure
+{
+    internal static class UriHelpers
+    {
+        public static bool IsSubdomainOf(Uri subdomain, Uri domain)
+        {
+            return subdomain.IsAbsoluteUri 
+                && domain.IsAbsoluteUri
+                && subdomain.Scheme == domain.Scheme
+                && subdomain.Port == domain.Port
+                && subdomain.Host.EndsWith($".{domain.Host}");
```

Comment:
Is the purpose of the subdomain check to allow *only* subdomains, or also to allow the domain itself? This updated check will fail for the domain itself (i.e. a domain is not a subdomain of itself).

I have a feeling that the more common case is to allow both subdomains and the domain itself. Thoughts?